pkg/services/star/startest/startest.go

```diff
@@ -0,0 +1,43 @@
+package startest
+
+import (
+	"context"
+
+	starmodel "github.com/grafana/grafana/pkg/services/star/model"
+)
+
+type FakeStarService struct {
+	ExpectedStars     *starmodel.Star
+	ExpectedError     error
+	ExpectedUserStars map[int64]bool
+}
+
+func NewStarServiceFake() *FakeStarService {
+	return &FakeStarService{}
+}
+
+func (f *FakeStarService) IsStarredByUserCtx(ctx context.Context, query *starmodel.IsStarredByUserQuery) (bool, error) {
+	return true, f.ExpectedError
+}
+
+func (f *FakeStarService) StarDashboard(ctx context.Context, cmd *starmodel.StarDashboardCommand) error {
+	return f.ExpectedError
+}
+
+func (f *FakeStarService) UnstarDashboard(ctx context.Context, cmd *starmodel.UnstarDashboardCommand) error {
+	return f.ExpectedError
+}
+
+func (f *FakeStarService) GetUserStars(ctx context.Context, query *starmodel.GetUserStarsQuery) (map[int64]bool, error) {
+	return f.ExpectedUserStars, f.ExpectedError
+}
+
+type FakeStarStore struct {
+	ExpectedStars     *starmodel.Star
+	ExpectedListStars []*starmodel.Star
+	ExpectedError     error
+}
+
+func NewStarStoreFake() *FakeStarStore {
+	return &FakeStarStore{}
+}
```

Comment:
What do you think about using `mockery` instead? 
https://github.com/grafana/grafana/blob/9465eb1b3abc9568898a8e298494e59ae27db416/pkg/services/dashboards/dashboard.go#L57

Reply:
I tend to prefer hand-crafted fakes over generated mocks, especially where they're provided from the maintainer of the service. The way [Testing on the Toilet: Know your test doubles](https://testing.googleblog.com/2013/07/testing-on-toilet-know-your-test-doubles.html) presents mocks vs fakes makes me think I'm not alone in that.

I think mocks are a decent tool to test how a service reacts to different hard-to-reproduce scenarios coming from a dependency, but overreliance on them risks making testing verbose and hard to debug.

Reply:
the agreement we reached last time is for now if we don't have complex test senario we use fake for instant, when complex senario appears later, we may have to mock.